App.tsx

```diff
@@ -1,14 +1,39 @@
+import { config } from "@tamagui/config/v2";
+import { Suspense, useState } from "react";
+import { Spinner, TamaguiProvider, YStack, createTamagui } from "tamagui";
+
 import { Header } from "./components/Header";
 import { TodoList } from "./components/TodoList";
 import { initializeLibraries } from "./init";
+import { wrapPromise } from "./utils/wrapSuspendPromise";
+import { TodoStore, loadFromStorage } from "./stores/Todo.store";
+import { NewTodoButton } from "./components/NewButton";
 
 initializeLibraries();
 
-export default function App() {
+const todoLoader = wrapPromise(loadFromStorage(), []);
+
+function LoadedApp() {
+  const todos = todoLoader.suspendUntilLoaded();
+  const [store] = useState(() => new TodoStore(todos ?? []));
```

Comment:
OK here's come a long comment.

Here comes another paint point with state management. You use the root App  component to host your global 'Store' state. It would be preferable that the store is initiated and hosted outside the React tree. It's a singleton (it lives as long as the app lives), so you could create an instance and export it from 'Store.ts'

```
export const store = new Store()
```

But then you have to hydrate the store from the async storage.
So it would be better to have the Store itself handle its hydration. 
Right now your App Component plays the orchestrator between storage and the Store. That is becoming spaghetti code.  
So it is preferable to trigger a 'loadTodos' from inside Todo's constructor.

Another pain point here is that if we want in the future to change the source of the hydration (eg. instead of asyncStore a backend API), then we would have to modify the App Component. Why? It should not be its concern. ;-). So this is the reason we write 'gateway' classes that abstract away the source of the data, and they just fetch them and deliver them to the stores (remember the marketLaunch structure)?

Now I understand that this will break the 'suspense'. Suspense is something 'react'-specific, and you see that the framework is already trying to 'trap' you in using its tools and methods. 

But we could break free from this. wrapPromise just wraps a promise. Mobx has an `when` async function that returns a promise.
So a good idea would be to have load and 'isLoading' in the Store class, and then write a suspendable promise that check these from the store and act as suspense client.

You're up to this challenge ;-)

Reply:
Okay, just making sure I understand:
- TodoGateway, the existing `loadFromStorage` logic
- TodoStore::load, calls gateway.load while managing this.isLoading and isLoaded
- Make a promise that wraps the isLoading boolean of TodoStore, and pass that to wrapPromise.

Reply:
Done, except I had to get rid of suspense in the meantime.